libresapi/src/api/FileSharingHandler.cpp

```diff
@@ -84,7 +84,9 @@ void FileSharingHandler::handleGetSharedDir(Request& /*req*/, Response& resp)
 	DirDetails dirDetails;
 	mRsFiles->RequestDirDetails(NULL, dirDetails, RS_FILE_HINTS_LOCAL);
 
+#pragma GCC diagnostic ignored "-Wstrict-aliasing"
 	resp.mDataStream << makeKeyValue("parent_reference", *reinterpret_cast<int*>(&dirDetails.ref));
```

Comment:
The proper way to fix this is not to hide the warning, but to cast using intptr_t:

resp.mDataStream << makeKeyValue("...", (int*)(intptr_t)(dirDetails.ref));

Reply:
Done but use:
*reinterpret_cast<int*>(reinterpret_cast<intptr_t>(&dirDetails.ref))
Yours failed to compiles.